lib/rialto/etl/writers/sparql_statement_writer.rb

```diff
@@ -27,42 +26,44 @@ def serialize(context)
         end
 
         # rubocop:disable Metrics/CyclomaticComplexity
+        # rubocop:disable Metrics/PerceivedComplexity
         def serialize_hash(hash, graph_name = nil)
           statements = []
-          subject_id = hash['@id'].to_s
-          subject = RDF::URI.new(hash['@id_ns'] + subject_id)
+          subject = RDF::URI.new(hash['@id'])
           graph_name ||= hash['@graph']
 
-          # Always add a valid date
-          statements << values_to_delete_insert(subject, Vocabs::DCTERMS['valid'], Time.now.to_date, graph_name, true)
-
           # Type
-          statements << values_to_delete_insert(subject, RDF.type, hash['@type'], graph_name, hash.key?('!type'))
-
-          # Label
-          # SKOS.prefLabel & VCARD.fn
-          statements << values_to_delete_insert(subject,
-                                                [Vocabs::SKOS['prefLabel'],
-                                                 Vocabs::FOAF['fn']], hash['@label'],
-                                                graph_name,
-                                                hash.key?('!label'))
-
-          # Person name vcard
-          statements << person_name_to_statements(subject, hash, graph_name) if hash.key?('@person_name')
-
-          # Person address vcard
-          statements << person_address_to_statements(subject, hash, graph_name) if hash.key?('@person_address')
+          statements << values_to_delete_insert(subject, RDF.type, hash['@type'], graph_name, true) if hash.key?('@type')
 
-          # Advisees
-          statements << advisees_to_statements(subject, subject_id, hash, graph_name) if hash.key?('@advisees') &&
-                                                                                         !hash['@advisees'].empty?
+          hash.each do |field, value|
+            # Ignore any @fields or !fields or #fields
+            next if field.start_with?('@', '!', '#')
+            values = value.is_a?(Array) ? value : [value]
+            # If value is not an array, make an array.
+            graph = RDF::Graph.new
+            values.compact.each do |this_value|
+              graph << if this_value.is_a?(Hash)
+                         statements << serialize_hash(this_value, graph_name)
+                         [subject, RDF::URI.new(field), RDF::URI.new(this_value['@id'])]
+                       else
+                         [subject, RDF::URI.new(field), this_value]
+                       end
+              # If !key, then delete.
+              statements << graph_to_delete([[subject, RDF::URI.new(field), nil]], graph_name) if hash.key?('!' + field)
```

Comment:
Just checking: did you mean to pass an array with an inner array here?

Reply:
Yes, that is a shortcut for a graph containing a triple.